File: 0119-pascals-triangle-ii/solution.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        if rowIndex == 0:
            return [1]
        elif rowIndex == 1:
            return [1, 1]
        else:
            prev_result = self.getRow(rowIndex - 1)
            result = [1]
            for i in range(len(prev_result)-1):
                result.append(prev_result[i] + prev_result[i+1])
            result.append(1)
            return result

File: 0119-pascals-triangle-ii/solution.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> pascal_row = {1};
        for (int cur_row = 1; cur_row <= rowIndex; cur_row++) {
            vector<int> next_row = {1};
            for (int col = 0; col < pascal_row.size() - 1; col++) {
                next_row.push_back(pascal_row[col] + pascal_row[col+1]);
            }
            next_row.push_back(1);
            pascal_row = std::move(next_row);
        }
        return pascal_row;
    }
};
